test/functional/p2p_mutated_blocks.py

```diff
@@ -74,7 +79,7 @@ def self_transfer_requested():
 
         # Attempt to clear the honest relayer's download request by sending the
         # mutated block (as the attacker).
-        with self.nodes[0].assert_debug_log(expected_msgs=["bad-txnmrklroot, hashMerkleRoot mismatch"]):
+        with self.nodes[0].assert_debug_log(expected_msgs=["Block mutated: bad-txnmrklroot, hashMerkleRoot mismatch"]):
```

Comment:
This is not necessary, right?

I added this assertion but asserting logs is not good practice: if it is really necessary then the interface under test should expose enough info to make the required assertion. Logs aren't a stable interface nor are they meant to serve as an interface for automated tests. Most of the time, duplicating them for test assertions just causes churn down the road. You don't want to force devs to "fix" a bunch of tests because they changed a user facing log message.

Reply:
It's not strictly necessary, no, but if we somehow added a regression that missed the mutation check and only caught it in the `AcceptBlock` portion, this change would catch it? I guess this could be shortened to just catching "Block mutated:" since that's the actual test(making sure it's not making further in validation)?

I am not strongly opinionated on it.

Reply:
> but if we somehow added a regression that missed the mutation check and only caught it in the AcceptBlock portion, this change would catch it?

Sure but testing fine grained implementation details like this sounds more like the job of a unit test to me.

Reply:
if we don't care about why we're disconnecting, we could just not log read at all and disconnect?